tests/tests_unit/test_utils/test_time.py

```diff
@@ -45,12 +45,19 @@ class TestDatetimeToMs:
             ("America/Los_Angeles", 1517385600000),
         ],
     )
-    def test_naive_datetime_to_ms(self, local_tz, expected_ms):
+    def test_naive_datetime_to_ms_unix(self, local_tz, expected_ms):
         with tmp_set_envvar("TZ", local_tz):
             time.tzset()
             assert datetime_to_ms(datetime(2018, 1, 31, tzinfo=None)) == expected_ms
             assert timestamp_to_ms(datetime(2018, 1, 31, tzinfo=None)) == expected_ms
 
+    @pytest.mark.skipif(
+        platform.system() != "Windows", reason="Windows assume naive dateimes are UTC, Unix assumes they are local"
```

Comment:
That’s not correct afaik. Python naive datetimes are always assumed local time

Reply:
?? Why are the tests passing then?

Reply:
There is a warning in the datetime docs stating: 

```
Warning: Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC.
```

It is  not clear to me whether _many_ datetime method means them all, or if there are some that treats them differently.

Reply:
@qTipTip : _All methods_ assume naive = local time, except those prefixed with `utc`. This makes for example `datetime.utcnow()` particularly dangerous, as it returns a `datetime` in UTC - but naive - so that future method calls e.g. `.timestamp()` will give the wrong result - unless your local time is set to UTC ofc.

Reply:
@doctrino : That must mean your system is using UTC or you have a setting somewhere that the Python interpreter picks up, for example, the environment variable `TZ` is set.

https://discuss.python.org/t/get-now-with-local-system-timezone-without-knowing-the-timezone-first/22665/12

Reply:
<img width="774" alt="image" src="https://user-images.githubusercontent.com/8521241/235900861-93c304e3-a38f-420c-9c53-109eb0e2e832.png">

Reply:
I booted up my window pc to try this out, and I also got the subtraction of two naive datetimes to be computed in "absolute" time, i.e. same as if both were given UTC timezone info. What's interesting though is that this works the same on unix.

Suggested fix: remove the platform check and let it run on both win&unix OR lets revert this method entirely (back to just `.timestamp()`. What do you think?

Reply:
### My final opinion
`datetime_to_ms` needs to give the same result for the same input for any OS / platform. This is a higher requirement than supporting naive datetimes before epoch on Windows, so these necessarily need to fail.

Reply:
Agreed to raise a ValueError instead with the message to not use native datetime.